### PR TITLE
Feature: TX-Builder - Make it work on xDai

### LINKED ISSUE #53: [xDai] - Transaction Builder ABI lookup
Parametrize the address to lookup for the ABI based on the connected network.

Currently, etherscan is hardcoded.

### PR DESCRIPTION
**Pull request by nicosampler**
Closes #53.

### CODE REVIEW

**Comment by nicosampler** [comment]
@dasanra should we request a new environment? 

* https://safe-apps.dev.gnosisdev.com
* https://safe-apps.staging.gnosisdev.com
* https://apps.gnosis-safe.io

None of these environments allow us to test the TX-builder on xDai. Another alternative is to remove the env var and handle it in the code as we are doing in safe-react (src/config), the drawback of it is that we need to hardcode the URLs (infura and https://rpc.xdaichain.com).

File: src/apps/TxBuilder/hooks/useServices/interfaceRepository.ts [hunk 1]
```diff
@@ -15,6 +16,24 @@ export interface ContractInterface {
 
 const getContract = memoize(async (apiUrl: string) => axios.get(apiUrl));
 
+const getAbiUrlByNetwork: {
```
Comment (fernandomg) [inline src/apps/TxBuilder/hooks/useServices/interfaceRepository.ts:19]: the functionality is there, I'm not so sure about the `get` part of the name for this map-like object.

Comment (nicosampler) [reply src/apps/TxBuilder/hooks/useServices/interfaceRepository.ts:19]: that's right, I thought it as a function then moved it into an object. any suggestion? 
removing the `get` perhaps?

Comment (fernandomg) [reply src/apps/TxBuilder/hooks/useServices/interfaceRepository.ts:19]: Yes.

In fact, what this returns is not an `abiUrlByNetwork` but a function that returns the `abiUrl`.

So, thinking in that direction I thought something around:
```typescript
const getAbiUrl = abiUrlGetterByNetwork[networkName]

if (getAbiUrl === null) {
  throw Error ('...')
}

const abi = getAbi(getAbiUrl(contractAddress))
```

would be most suitable.

But I'm not the master-on-naming-things :mage_man:, so, your call.

Comment (nicosampler) [reply src/apps/TxBuilder/hooks/useServices/interfaceRepository.ts]: done! thx for the detailed feedback :)

**Review by fernandomg — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
   "private": true,
   "dependencies": {
     "@apollo/react-hooks": "^3.1.3",
-    "@gnosis.pm/safe-apps-sdk": "0.4.1",
+    "@gnosis.pm/safe-apps-sdk": "https://github.com/gnosis/safe-apps-sdk.git#ea8dcab",
```
Comment (dasanra) [inline package.json:7]: v0.4.2 is being released right now with the necessary changes. I would use that version.

File: src/apps/TxBuilder/hooks/useServices/interfaceRepository.ts [hunk 3]
```diff
@@ -25,19 +44,17 @@ class InterfaceRepository {
   }
 
   private async _loadAbiFromEtherscan(address: string): Promise<string> {
```
Comment (dasanra) [inline src/apps/TxBuilder/hooks/useServices/interfaceRepository.ts:46]: `_loadAbiFromBlockExplorer`?

File: package.json (hunk 2 above)
Comment (nicosampler) [reply package.json:7]: it's not yet on npm

File: src/apps/TxBuilder/hooks/useServices/interfaceRepository.ts (hunk 3 above)
Comment (nicosampler) [reply src/apps/TxBuilder/hooks/useServices/interfaceRepository.ts]: done!

File: package.json (hunk 2 above)
Comment (nicosampler) [reply package.json]: @dasanra done!

**Review by dasanra — APPROVED** [review]

**Pull request merged**